application/src/main/kotlin/net/corda/v5/application/crypto/DefaultSignatureSpecService.kt

```diff
@@ -0,0 +1,26 @@
+package net.corda.v5.application.crypto
+
+import net.corda.v5.crypto.SignatureSpec
+import java.security.PublicKey
+
+interface DefaultSignatureSpecService {
+    /**
+     * Work out a signature and hash algorithm to use for a given public key, given current security policies.
+     *
+     * @param publicKey the public key to be used for signing
+     *
+     * @return An appropriate [SignatureSpec], or null if nothing is available for the key type.
+     */
+    fun suggest(publicKey: PublicKey): SignatureSpec?
+
+    /**
+     * Work out a signature algorithm given current security policies and a hash algorithm
+     *
+     * @param publicKey the public key to be used for signing
+     * @param hash the hash algortithm e.g. "SHA256" - TODO: use an interface or enum here?
+     *
+     * @return An appropriate [SignatureSpec], or null if nothing is available for the key type
+     */
+
+    fun suggest(publicKey: PublicKey, hash: String): SignatureSpec?
```

Comment:
Needs @Suspendable. hash parameters should be `DigestAlgorithmName`

Reply:
Done